src/module/token/royalty/RoyaltyERC1155.sol

```diff
@@ -1,32 +1,304 @@
+            FallbackFunction({selector: this.getTransferValidationFunction.selector, permissionBits: 0});
+        config.fallbackFunctions[5] =
+            FallbackFunction({selector: this.setDefaultRoyaltyInfo.selector, permissionBits: Role._MANAGER_ROLE});
+        config.fallbackFunctions[6] =
             FallbackFunction({selector: this.setRoyaltyInfoForToken.selector, permissionBits: Role._MANAGER_ROLE});
-
-        config.supportedInterfaces = new bytes4[](1);
-        config.supportedInterfaces[0] = 0x2a55205a; // IERC2981.
+        config.fallbackFunctions[7] =
+            FallbackFunction({selector: this.setTransferValidator.selector, permissionBits: Role._MANAGER_ROLE});
 
         config.requiredInterfaces = new bytes4[](1);
         config.requiredInterfaces[0] = 0xd9b67a26; // ERC1155
 
+        config.supportedInterfaces = new bytes4[](1);
+        config.supportedInterfaces[0] = 0x2a55205a; // IERC2981.
+
         config.registerInstallationCallback = true;
     }
 
+    /*//////////////////////////////////////////////////////////////
+                    Encode install / uninstall data
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Returns bytes encoded install params, to be sent to `onInstall` function
+    function encodeBytesOnInstall(address royaltyRecipient, uint256 royaltyBps, address transferValidator)
+        external
+        pure
+        returns (bytes memory)
+    {
+        return abi.encode(royaltyRecipient, royaltyBps, transferValidator);
+    }
+
+    /// @dev Returns bytes encoded uninstall params, to be sent to `onUninstall` function
+    function encodeBytesOnUninstall() external pure returns (bytes memory) {
+        return "";
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            CALLBACK FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Callback function for ERC1155.transferFrom/safeTransferFrom
+    function beforeTransferERC1155(address from, address to, uint256 _id, uint256 _value)
+        external
+        virtual
+        override
+        returns (bytes memory)
+    {
+        address transferValidator = getTransferValidator();
+        if (transferValidator != address(0)) {
+            ITransferValidator(transferValidator).validateTransfer(msg.sender, from, to, _id, _value);
+        }
+    }
+
+    function beforeBatchTransferERC1155(address from, address to, uint256[] calldata ids, uint256[] calldata values)
+        external
+        virtual
+        override
+        returns (bytes memory)
+    {
+        address transferValidator = getTransferValidator();
+        if (transferValidator != address(0)) {
+            uint256 length = ids.length;
+            for (uint256 i = 0; i < length;) {
+                ITransferValidator(transferValidator).validateTransfer(msg.sender, from, to, ids[i], values[i]);
+                unchecked {
```

Comment:
i believe this is obsolete in newer solidity version: https://soliditylang.org/blog/2023/10/25/solidity-0.8.22-release-announcement/